crates/goose/src/agent.rs

```diff
@@ -5,6 +5,7 @@ use rust_decimal_macros::dec;
 use serde_json::json;
 use std::collections::HashMap;
 use tokio::sync::Mutex;
+use tracing::{debug, instrument, warn};
```

Comment:
will have some conflicts with this file with the mcp changes, but we can forward port easily i think